tests_extractor.d

```diff
@@ -61,8 +61,9 @@ class TestVisitor : ASTVisitor
         {
             outFile.writeln(q{extern(C) void main()
 {
-    static foreach(u; __traits(getUnitTests, __traits(parent, main)))
-        u();
+    static foreach(test; __traits(allMembers, mixin(__MODULE__)))
+        static if (test != "main" && is(typeof(mixin(test)())))
+            mixin(test)();
```

Comment:
Hmm, this is less cleaner than iterating over unittest.
Maybe we can at least do a check whether it starts with unittest to avoid the main special case?
So something like:

```d
test.length > 8 && test[0..8] == "unittest"
```

Or strcmp if that's not possible in betterC.

This way we could get rid of the check for callability as well.

Reply:
done